torch/distributed/_shard/sharded_tensor/_ops/math_ops.py

```diff
@@ -49,7 +61,14 @@ def binary_math_op(types, args=(), kwargs=None, pg=None):
 
         elif isinstance(rhs, ReplicatedTensor):
             assert isinstance(lhs, ShardedTensor)
-            res = op(lhs.local_tensor(), rhs)
+            st_size = lhs.size()
+            st_meta = lhs.local_shards()[0].metadata
+            if st_size != lhs.size():
+                # try to broadcast replicated tensor
+                rhs.expand(st_size)
```

Comment:
Wondering why `test_replicated_tensor_inter_op_sharded_tensor` didn't catch this issue? Or am I missing something here?

Reply:
just fixed this, the `test_replicated_tensor_inter_op_sharded_tensor` was not triggering bc it does not trigger implicit broadcasting, I fixed the `test_replicated_tensor_implicit_broadcasting` and use size of (3,) for replicatedtensor, locally verified it triggers the failure if we don't use the return value of expand.

Reply:
Hmm, so why doesn't (1, 3) trigger broadcasting as well? It should right?

Reply:
For the case of (1, 3), it right now should correctly trigger the `expand` logic as well. The reason it was not triggered is because:
* There were some logic wrong to trigger the `expand` call, I used the wrong operands to compare the size
* even if it was not getting expanded, `narrow_tensor` call still returns the original tensor instead of throwing as it's too small to narrow on dim 0
* so `res = op(lhs.local_tensor(), replica_part)` got executed with size of `op((3, 3), (1, 3))`, where the implicit broadcasting happened in local shard computation